src/doc/DocAnnotator.js

```diff
@@ -1031,22 +985,36 @@ class DocAnnotator extends Annotator {
      * @return {void}
      */
     handleControllerEvents(data) {
-        const isCreateDialogVisible = this.createHighlightDialog && this.createHighlightDialog.isVisible;
-
         switch (data.event) {
             case CONTROLLER_EVENT.toggleMode:
-                if (isCreateDialogVisible) {
-                    document.getSelection().removeAllRanges();
-                    this.createHighlightDialog.hide();
-                }
+                document.getSelection().removeAllRanges();
+                this.modeControllers[TYPES.highlight].createDialog.hide();
+                this.modeControllers[TYPES.highlight_comment].createDialog.hide();
```

Comment:
The above comment, about abstracting this, can have guards to make these a little bit safer

Reply:
Actually moved this into the `annotator.toggleAnnotationMethod()` because it needs to occur on every mode not just highlights